src/clients/api.ts

```diff
@@ -0,0 +1,235 @@
+import * as t from "io-ts";
+
+// A basic response type that also include 401
+import {
+  ApiHeaderJson,
+  basicErrorResponseDecoder,
+  basicResponseDecoder,
+  BasicResponseType,
+  composeHeaderProducers,
+  composeResponseDecoders,
+  createFetchRequestForApi,
+  IGetApiRequestType,
+  IPostApiRequestType,
+  IResponseType,
+  RequestHeaderProducer,
+  ResponseDecoder
+} from "italia-ts-commons/lib/requests";
+import nodeFetch from "node-fetch";
+import { ExtendedProfile } from "../types/api/ExtendedProfile";
+import { LimitedProfile } from "../types/api/LimitedProfile";
+import { MessageResponseWithContent } from "../types/api/MessageResponseWithContent";
+import { Messages } from "../types/api/Messages";
+import { ServicePublic } from "../types/api/ServicePublic";
+import { Services } from "../types/api/Services";
+
+const OcpApimSubscriptionKey = "Ocp-Apim-Subscription-Key";
+type OcpApimSubscriptionKey = t.TypeOf<typeof OcpApimSubscriptionKey>;
+
+// ProfileLimitedOrExtended is oneOf [LimitedProfile, ExtendedProfile]
+const ProfileLimitedOrExtended = t.union([LimitedProfile, ExtendedProfile]);
+
+export type ProfileLimitedOrExtended = t.TypeOf<
+  typeof ProfileLimitedOrExtended
+>;
+
+export type BasicResponseTypeWith401<R> =
```

Comment:
`BasicResponseType` is enough, the CD API never responds with a 401 (see [the specs](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/api/public_api_v1.yaml))

Reply:
what's the status in case the API gateway cannot authenticate the request ?

Reply:
ah yes, is it supposed to be defined in the specs?